pkg/apis/discovery/validation/validation.go

```diff
@@ -83,17 +83,20 @@ func validateEndpoints(endpoints []discovery.Endpoint, addrType discovery.Addres
 		idxPath := fldPath.Index(i)
 		addressPath := idxPath.Child("addresses")
 
-		if addrType == discovery.AddressTypeIP {
-			if len(endpoint.Addresses) == 0 {
-				allErrs = append(allErrs, field.Required(addressPath, "must contain at least 1 address"))
-			} else if len(endpoint.Addresses) > maxAddresses {
-				allErrs = append(allErrs, field.TooMany(addressPath, len(endpoint.Addresses), maxAddresses))
-			}
+		if len(endpoint.Addresses) == 0 {
+			allErrs = append(allErrs, field.Required(addressPath, "must contain at least 1 address"))
+		} else if len(endpoint.Addresses) > maxAddresses {
+			allErrs = append(allErrs, field.TooMany(addressPath, len(endpoint.Addresses), maxAddresses))
+		}
 
-			for i, address := range endpoint.Addresses {
+		for i, address := range endpoint.Addresses {
+			switch addrType {
```

Comment:
my reading of the caller is that it'd be possible to get here with an addrType of "", which sidesteps all this checking -- should probably either test or fix that.

Reply:
We do have defaulting in place here that should default this to IP as an address type when it isn't specified. Above this we also have validation that ensures that the `addressType` is one of the `supportedAddressTypes`. So as long as every entry in `supportedAddressTypes` is also covered here, we should be covered. Though of course this could all change if it makes sense to make this validation less restrictive to simplify supporting additional address types in the future.

Reply:
https://github.com/kubernetes/kubernetes/pull/84091/files#diff-d3dbc1e6a66cbe3fe6f1a843d112b96dL48-L50

I assume defaulting causes you to never take the first branch in the if statement, but if we're so confident in that, why would we even have it?

Reply:
Yep, you're completely right. It just felt wrong to leave that unguarded without a nil check, but it is unnecessary so I've removed it.

Reply:
Yeah, that feels pretty weird, I'd actually recommend keeping the nil check (and just setting addrType to something valid if it fails).

Reply:
Sounds good. I've added the nil check back and added a comment with it to indicate that a default value should have already been set at that point.